pipeline/outputs/s3.md

```diff
@@ -1,141 +1,206 @@
 
-* `$TAG[0]` = my\_app\_name
-* `$TAG[1]` = logs
-* `$TAG[2]` = prod
+- `$TAG[0]` = `my_app_name`
+- `$TAG[1]` = `logs`
+- `$TAG[2]` = `prod`
 
-So the key in S3 will be `/prod/my_app_name/2020/01/01/00/00/00/bgdHN1NM.gz`.
+The key in S3 will be `/prod/my_app_name/2020/01/01/00/00/00/bgdHN1NM.gz`.
 
 ### Allowing a file extension in the S3 Key Format with $UUID
 
-The Fluent Bit S3 output was designed to ensure that previous uploads will never be over-written by a subsequent upload. Consequently, the `s3_key_format` supports time formatters, `$UUID`, and `$INDEX`. `$INDEX` is special because it is saved in the `store_dir`; if you restart Fluent Bit with the same disk, then it can continue incrementing the index from its last value in the previous run.
+The Fluent Bit S3 output was designed to ensure that previous uploads will never be
+over-written by a subsequent upload. The `s3_key_format` supports time formatters,
+`$UUID`, and `$INDEX`. `$INDEX` is special because it is saved in the `store_dir`; if
+you restart Fluent Bit with the same disk, then it can continue incrementing the
+index from its last value in the previous run.
 
-For files uploaded with the PutObject API, the S3 output requires that a unique random string be present in the S3 key. This is because many of the use cases for PutObject uploads involve a short time period between uploads such that a timestamp in the S3 key may not be unique enough between uploads. For example, if you only specify minute granularity timestamps in the S3 key, with a small upload size, it is possible to have two uploads that have timestamps set in the same minute. This "requirement" can be disabled with `static_file_path On`.
+For files uploaded with the `PutObject` API, the S3 output requires that a unique
+random string be present in the S3 key. Many of the use cases for
+`PutObject` uploads involve a short time period between uploads, so a timestamp
+in the S3 key may not be unique enough between uploads. For example, if you only
+specify minute granularity timestamps in the S3 key, with a small upload size, it is
+possible to have two uploads that have timestamps set in the same minute. This
+requirement can be disabled with `static_file_path On`.
 
-There are three cases where the PutObject API is used:
+There are three cases where the `PutObject` API is used:
 
-1. When you explicitly set `use_put_object On`
-2. On startup when the S3 output finds old buffer files in the `store_dir` from a previous run and attempts to send all of them at once.
-3. On shutdown, when to prevent data loss the S3 output attempts to send all currently buffered data at once.
+1. When you explicitly set `use_put_object On`.
+1. On startup when the S3 output finds old buffer files in the `store_dir` from
+   a previous run and attempts to send all of them at once.
+1. On shutdown. To prevent data loss the S3 output attempts to send all currently
+   buffered data at once.
```

Comment:
```suggestion
- When you explicitly set `use_put_object On`.
- On startup when the S3 output finds old buffer files in the `store_dir` from
  a previous run and attempts to send all of them at once.
- On shutdown. To prevent data loss the S3 output attempts to send all currently
  buffered data at once.
```